server/middleware.ts

```diff
@@ -171,7 +171,7 @@ export async function canUserModify(request: express.Request, response: express.
 			});
 			return;
 		}
-		if (user.applied && branchName.toLowerCase() !== user.applicationBranch.toLowerCase()) {
+		if (user.applied && branchName.toLowerCase() !== user.applicationBranch!.toLowerCase()) {
```

Comment:
Not sure about what's happens in the branch creation code but it might be a good idea to also `.trim()` `branchName` and `user.applicationBranch` just in case.

Reply:
I agree this code is a bit convoluted but:
`branchName` is produced by `BranchConfig.getCanonicalName()` which always returns a lower case string. This call to `.toLowerCase()` is emphasizing that it must be lower case. The `user.applicationBranch` value is set by registration and not any pesky users who like to insert extra spaces so it should be fine here as-is